testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -8897,56 +8898,44 @@ private void testCurrentDateFunc(Pair<String, Hook.Closeable> pair) {
     f0.forEachLibrary(list(SqlLibrary.BIG_QUERY, SqlLibrary.ORACLE), consumer);
   }
 
-  @Test void testStartsWithFunction() {
+  /** Tests the {@code STARTS_WITH} and {@code STARTSWITH} operator. */
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("startsWithAliases")
+  void testStartsWithFunction2(FunctionAlias functionAlias) {
     final SqlOperatorFixture f0 = fixture();
-    f0.setFor(SqlLibraryOperators.STARTS_WITH);
+    final SqlFunction function = functionAlias.function;
+    f0.setFor(function);
+    final String alias = functionAlias.alias;
     final Consumer<SqlOperatorFixture> consumer = f -> {
-      f.checkBoolean("starts_with('12345', '123')", true);
-      f.checkBoolean("starts_with('12345', '1243')", false);
-      f.checkBoolean("starts_with(x'11', x'11')", true);
-      f.checkBoolean("starts_with(x'112211', x'33')", false);
-      f.checkFails("^starts_with('aabbcc', x'aa')^",
-          "Cannot apply 'STARTS_WITH' to arguments of type "
-              + "'STARTS_WITH\\(<CHAR\\(6\\)>, <BINARY\\(1\\)>\\)'\\. Supported "
-              + "form\\(s\\): 'STARTS_WITH\\(<STRING>, <STRING>\\)'",
+      f.checkBoolean(alias + "('12345', '123')", true);
```

Comment:
Another possible suggestion is to have the alias implement a "rewrite" method which rewrites a specific template. Then you can write tests like `f.checkBoolean(alias.rewrite("%function%('12345', '1234')"))`. But I don't know if the readability of the test improves much.

Reply:
Do we need then `"%function%`?
May be just name this method `for` or `applyFor` and use it like
```java
f.checkBoolean(alias.applyFor("('12345', '1234')"));
```
?

Reply:
Your proposal works for standard function syntax, but sql is a weird language. But your proposal may be good enough for most use cases.

Reply:
i put it in a separate commit here to see how it could look like
https://github.com/apache/calcite/pull/3590/commits/88fc66326964d1019de37b9b924445ac5648ad6e